src/uu/comm/src/comm.rs

```diff
@@ -60,7 +70,71 @@ impl LineReader {
     }
 }
 
-fn comm(a: &mut LineReader, b: &mut LineReader, delim: &str, opts: &ArgMatches) {
+impl OrderChecker {
+    fn new(file_num: usize, check_order: bool) -> Self {
+        Self {
+            last_line: Vec::new(),
+            file_num,
+            check_order,
+            has_error: false,
+        }
+    }
+
+    fn check_order(&mut self, current_line: &[u8]) -> bool {
+        if self.last_line.is_empty() {
+            self.last_line = current_line.to_vec();
+            return true;
+        }
+
+        let is_ordered = current_line >= &self.last_line;
+        if !is_ordered && !self.has_error {
+            eprintln!("comm: file {} is not in sorted order", self.file_num);
+            self.has_error = true;
+        }
+
+        self.last_line = current_line.to_vec();
+        is_ordered || !self.check_order
```

Comment:
I'm struggling with the `||  !self.check_order` part. I think this functionality should be somewhere else.

Reply:
not sure how to fix that?! sorry